app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java

```diff
@@ -456,4 +459,48 @@ public static void startYourself(Context context) {
         Intent intent = new Intent(context, LoginActivity.class);
         context.startActivity(intent);
     }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        // if progressDialog is visible during the configuration change  then store state as  true else false so that
+        // we maintain visiblity of progressDailog after configuration change
+        if(progressDialog!=null&&progressDialog.isShowing()){
+            outState.putBoolean(Save_progressDailog,true);
+        }else{
+            outState.putBoolean(Save_progressDailog,false);
+        }
+
+        outState.putString(Save_errorMessage,errorMessage.getText().toString()); //Save the errorMessage
+        outState.putString(Save_username,getUsername()); // Save the username
+        outState.putString(Save_password,getPassword()); // Save thte password
+    }
+
+    private String getUsername() {
+        return usernameEdit.getText().toString();
+    }
+    private String getPassword(){
+        return  passwordEdit.getText().toString();
+  }
+
+    @Override
+    protected void onRestoreInstanceState(final Bundle savedInstanceState) {
+        super.onRestoreInstanceState(savedInstanceState);
+
+        usernameEdit.setText(savedInstanceState.getString(Save_username));
+        passwordEdit.setText(savedInstanceState.getString(Save_password));
+
+
+        if(savedInstanceState.getBoolean(Save_progressDailog)){
+            performLogin();
+        }
+
+        String errorMessage=savedInstanceState.getString(Save_errorMessage);
+
+        if(sessionManager.isUserLoggedIn()){
+            showMessage(R.string.login_success, R.color.primaryDarkColor);
+        }else{
```

Comment:
Please refer to codestyle, there should be whitespaces here and in other `else` statements.